### DISCUSSION TITLE
How to identify scanned pdf

### DISCUSSION THREAD

**Discussion by swathy-z9q** [pymupdf/PyMuPDF#1653] [Looking for help] [answered]
Hi, 

I require ideas on how to identify a pdf as scanned. 

I came across a pdf, where a scanned image was converted to pdf by a producer "[www.ilovepdf.com](http://www.ilovepdf.com/)" and have also identified many other producers as well which does OCR on scanned images and then converts to pdfs. This pdf file when opened by Fitz consists of text content generated by the producer, hence looks similar to proper native pdf or pdf generated from the word document(Not scanned pdfs). 

I need help in classifying which pdf is scanned or native. Is there any way to use pdf metadata or please suggest your ideas

Thanks in advance

**Comment by JorjMcKie** [comment, marked as answer]
There is no failsafe answer to this. But a few heuristics can at least help:
1. A scanned page will presumably be completely covered by an image (or two images - depends on the scanner used). So you can look at the images on the page and compare its / their rectangle(s) with the page rectangle. It may happen, that the image rectangle is not exactly equal to the page rectangle. So do not check for equality, but allow for some deviation like `abs(image_bbox & page.rect) / abs(page.rect) >= 0.95`. This means that the intersection area of image and page should at least cover 95% of the page ... you get the idea.
2. If text can be extracted and Tesseract was used for OCR, then a specific fontname, "GlyphlessFont", will be found in text extraction using `page.get_text("dict")`. Should also be found looking at `page.get_fonts()`.
3. Depending on the scanner, some information may also be put into PDF metadata. This obviously is highly dependant on the product used.

**Reply by swathy-z9q** [reply]
Thanks a lot for sharing these approaches :+1:
